sw/device/mask_rom/mask_rom.c

```diff
@@ -3,13 +3,39 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "sw/device/mask_rom/mask_rom.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
 #include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/rom_exts/rom_ext_manifest_parser.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// This is the expected ROM_EXT Manifest Identifier
+static uint32_t const kRomExtIdentifierExpected = 0x4552544F;
+
+typedef void (*boot_fn)(void);
 
 void mask_rom_boot(void) {
-  asm volatile("unimp");
-  __builtin_unreachable();
+  rom_ext_manifest_t rom_ext = rom_ext_get_parameters(kRomExtManifestSlotA);
+
+  // Check we have a valid ROM_EXT
+  if (rom_ext_get_identifier(rom_ext) == kRomExtIdentifierExpected) {
+    // Set mtvec for ROM_EXT and jump to entry point.
+    asm volatile("csrw mtvec, %0" ::"r"(rom_ext.slot + 0x400u));
```

Comment:
These will eventually be part of the parser, but the parser doesn't yet have accessors for these fields.

Reply:
I've added these to the parser, but as we're close to landing the generator that may cause issues because these offsets are not yet generated by the generator (as they're not really fields like the other pieces are).

Reply:
I've added these to the parser in #3800.